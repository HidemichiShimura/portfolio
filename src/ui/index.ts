// Atoms
export { default as PageHeading } from "./atoms/PageHeading/PageHeading";
export { default as Avatar } from "./atoms/Avatar/Avatar";
export { default as Typography } from "./atoms/Typography/Typography";
export { default as SectionHeading } from "./atoms/SectionHeading/SectionHeading";
export { default as Icon } from "./atoms/Icon/Icon";
export { default as RadialLinearContainer } from "./atoms/RadialLinearContainer/RadialLinearContainer";
export { default as Grid } from "./atoms/Grid/Grid";
export { default as GradientCard } from "./atoms/GradientCard/GradientCard";
export { default as Card } from "./atoms/Card/Card";

// Molecules
export { default as HeroHeadings } from "./molecules/HeroHeadings/HeroHeadings";
