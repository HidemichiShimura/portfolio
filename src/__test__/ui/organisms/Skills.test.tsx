import { Skills } from "@/ui";
import { screen, render } from "@testing-library/react";

const TITLE_ATTR = "Skills";
const SECTION_TITLE = "SKILLS";

describe("Organisms: <Skills />", () => {
  test("render", () => {
    render(<Skills />);

    expect(screen.getByTitle(TITLE_ATTR)).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    render(<Skills />);

    expect(screen.getByTitle(TITLE_ATTR)).toMatchSnapshot();
  });

  test("contains a section title", () => {
    render(<Skills />);

    expect(screen.queryByRole("heading")).toBeInTheDocument();
  });

  test("displays the correct section title", () => {
    render(<Skills />);

    expect(screen.queryByRole("heading")?.textContent).toBe(SECTION_TITLE);
  });
});
