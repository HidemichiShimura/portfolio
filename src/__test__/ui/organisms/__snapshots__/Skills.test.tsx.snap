// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms: <Skills /> matches snapshot 1`] = `
<section
  class="flex min-h-screen flex-col justify-center gap-8"
  title="Skills"
>
  <h2
    class="bg-gradient-to-b from-black to-neutral-100 bg-clip-text font-sans text-4xl text-transparent dark:from-neutral-100 dark:to-black"
  >
    SKILLS
  </h2>
  <div
    class="grid grid-cols-1 grid-rows-5 gap-4 md:grid-cols-6 md:grid-rows-3"
  >
    <div
      class="col-span-1 row-span-1 h-full md:col-span-2 md:row-span-2"
    >
      <div
        class="min-h-full min-w-full rounded-lg bg-[linear-gradient(35deg,rgba(121,82,179,0.5)_0%,transparent_25%,transparent_75%,rgba(121,82,179,0.5)_100%)]"
      >
        <div
          class="flex min-h-full flex-col gap-2 p-2"
        >
          <p
            class="text-light-gray text-base font-sans"
          >
            Languages
          </p>
          <div
            class="flex items-center justify-center md:h-36"
          >
            <ul
              class="flex w-4/5 items-center justify-center gap-2 "
            >
              <li
                class="min-w-8 md:w-16"
              >
                <img
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8 md:w-16"
              >
                <img
                  src="/img.jpg"
                />
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-span-1 row-span-1 md:col-span-4 md:row-span-1"
    >
      <div
        class="min-h-full min-w-full rounded-lg bg-[linear-gradient(35deg,rgba(121,82,179,0.5)_0%,transparent_25%,transparent_75%,rgba(121,82,179,0.5)_100%)]"
      >
        <div
          class="flex min-h-full flex-col gap-2 p-2"
        >
          <p
            class="text-light-gray text-base font-sans"
          >
            Frontend
          </p>
          <div
            class="flex items-center justify-center"
          >
            <ul
              class="flex w-4/5 items-center justify-center gap-2 "
            >
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  class="dark:invert"
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-span-1 row-span-1 md:col-span-4 md:row-span-1"
    >
      <div
        class="min-h-full min-w-full rounded-lg bg-[linear-gradient(35deg,rgba(121,82,179,0.5)_0%,transparent_25%,transparent_75%,rgba(121,82,179,0.5)_100%)]"
      >
        <div
          class="flex min-h-full flex-col gap-2 p-2"
        >
          <p
            class="text-light-gray text-base font-sans"
          >
            Backend
          </p>
          <div
            class="flex items-center justify-center"
          >
            <ul
              class="flex w-4/5 items-center justify-center gap-2 "
            >
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  class="dark:invert"
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-span-1 row-span-1 md:col-span-4 md:row-span-1"
    >
      <div
        class="min-h-full min-w-full rounded-lg bg-[linear-gradient(35deg,rgba(121,82,179,0.5)_0%,transparent_25%,transparent_75%,rgba(121,82,179,0.5)_100%)]"
      >
        <div
          class="flex min-h-full flex-col gap-2 p-2"
        >
          <p
            class="text-light-gray text-base font-sans"
          >
            Tools
          </p>
          <div
            class="flex items-center justify-center"
          >
            <ul
              class="flex w-4/5 items-center justify-center gap-2 "
            >
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  class="invert"
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
              <li
                class="min-w-8"
              >
                <img
                  src="/img.jpg"
                />
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-span-1 row-span-1 md:col-span-2 md:row-span-1"
    >
      <div
        class="min-h-full min-w-full rounded-lg bg-[linear-gradient(35deg,rgba(121,82,179,0.5)_0%,transparent_25%,transparent_75%,rgba(121,82,179,0.5)_100%)]"
      >
        <div
          class="flex h-20 items-center justify-center"
        >
          <p
            class="text-light-gray text-base font-sans"
          >
            Stay tuned...
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;
